dialect/sql/schema/postgres.go

```diff
@@ -272,7 +272,7 @@ func (d *Postgres) scanColumn(c *Column, rows *sql.Rows) error {
 			schemaType := fmt.Sprintf("varchar(%d)", characterMaximumLen.Int64)
 			c.SchemaType = map[string]string{dialect.Postgres: schemaType}
 		}
-	case "date", "time", "timestamp", "timestamp with time zone", "timestamp without time zone":
+	case "date", "time with time zone", "time without time zone", "timestamp with time zone", "timestamp without time zone":
```

Comment:
```suggestion
	case "date", "time", "timestamp", "time with time zone", "time without time zone", "timestamp with time zone", "timestamp without time zone":
```

Please add back the "time" and "timestamp" types as they are also valid inputs.

Reply:
Thank you for your review.
Is it necessary?

```go
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Int("age").Positive(),
		field.String("name").Default("unknown"),
		field.Time("field1").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time",
			}),
		field.Time("field2").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "timestamp",
			}),
		field.Time("field3").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time with time zone",
			}),
		field.Time("field4").Optional().
			SchemaType(map[string]string{
				dialect.Postgres: "time without time zone",
			}),
	}
}
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id")) args=[]
```

```sql
postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

`time` ➡️ `time without time zone`
`timestamp` ➡️ `timestamp without time zone`

Reply:
Can you verify this on Postgres 11, 12, 13, and 14?

I don't want to bother you, but I usually double-check myself before removing some code. Thanks

Reply:
I've confirmed it.

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 110014
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 120009
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 130005
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```

```sql
Exec: query=CREATE TABLE IF NOT EXISTS "users"("id" bigint GENERATED BY DEFAULT AS IDENTITY NOT NULL, "age" bigint NOT NULL, "name" varchar NOT NULL DEFAULT 'unknown', "field1" time NULL, "field2" timestamp NULL, "field3" time with time zone NULL, "field4" time without time zone NULL, PRIMARY KEY("id"))
```

```sql
postgres=# SHOW server_version_num;
 server_version_num
--------------------
 140001
(1 row)

postgres=# \d users
                                      Table "public.users"
 Column |            Type             | Collation | Nullable |             Default
--------+-----------------------------+-----------+----------+----------------------------------
 id     | bigint                      |           | not null | generated by default as identity
 age    | bigint                      |           | not null |
 name   | character varying           |           | not null | 'unknown'::character varying
 field1 | time without time zone      |           |          |
 field2 | timestamp without time zone |           |          |
 field3 | time with time zone         |           |          |
 field4 | time without time zone      |           |          |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
```